07_errors/sirigithub/dbstore.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+type dbStore struct {
+	dbfilePath string
+	currID     int
+}
+
+func NewdbStore(filePath string) *dbStore {
+	var store = dbStore{}
+	store.dbfilePath = filePath
+	return &store
+}
+func (s *dbStore) CreatePuppy(puppy *Puppy) (int, error) {
+	if puppy.Value < 0 {
+		return -1, NewError(ErrInvalidValue, "Puppy value must be greater than 0")
+	}
+	s.currID++
+	puppy.ID = s.currID
+	db := dbConn(s.dbfilePath)
+	defer db.Close()
+
+	value := getPuppyData(puppy)
+	err := db.Put([]byte(strconv.Itoa(puppy.ID)), value, nil)
+	if err != nil {
+		return -1, NewError(ErrDatabaseWrite, fmt.Sprintf("Error creating Puppy %d in the dbStore", puppy.ID))
```

Comment:
Generally you always want to wrap errors so you don't lose any detail so I'd write `NewError()` to take third `error` parameter (which can be nil) and store that error in your error type. See more detail below.